src/utils.rs

```diff
@@ -59,3 +60,15 @@ pub fn bytecode_to_code_hash_for_mode<const N: usize, E: VmEncodingMode<N>>(
 
     Ok(versioned_hash_bytes)
 }
+
+/// Erase start and page number from a fat pointer. To be used in the case of a fat pointer
+/// being passed to an opcode which shouldn't receive one.
+pub fn erase_fat_pointer_metadata(ptr: &mut U256) {
+    // Memory page is at 1<<32 - 1<<64, start is at 1<<64 - 1<<96
+    // We also need to preserve the top 128 bits of the value
+    let low_bits = ptr.low_u128() & 0xffffffff0000000000000000ffffffffu128;
```

Comment:
```suggestion
    let low_bits = 0xffffffff0000000000000000ffffffffu128;
```
We don't need to do bitwise and here since it is done at the end anyway. Maybe change the variable name from `tmp` to `mask` would make things clear.